android/gcm/app/src/main/java/gcm/play/android/samples/com/gcmquickstart/MainActivity.java

```diff
@@ -74,13 +81,15 @@ public void onReceive(Context context, Intent intent) {
     @Override
     protected void onResume() {
         super.onResume();
-        LocalBroadcastManager.getInstance(this).registerReceiver(mRegistrationBroadcastReceiver,
-                new IntentFilter(QuickstartPreferences.REGISTRATION_COMPLETE));
+
     }
 
     @Override
     protected void onPause() {
-        LocalBroadcastManager.getInstance(this).unregisterReceiver(mRegistrationBroadcastReceiver);
+        if(isReceiverRegistered) {
```

Comment:
No need for if here, since registration will occur either in onCreate or onResume, when onPause is called the receiver will be registered.